lib/srv/discovery/discovery.go

```diff
@@ -1749,9 +1765,17 @@ func (s *Server) upsertDynamicMatchers(ctx context.Context, dc *discoveryconfig.
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	s.muDynamicTAGSyncFetchers.Lock()
-	s.dynamicTAGSyncFetchers[dc.GetName()] = awsSyncMatchers
-	s.muDynamicTAGSyncFetchers.Unlock()
+	s.muDynamicTAGAWSFetchers.Lock()
+	s.dynamicTAGAWSFetchers[dc.GetName()] = awsSyncMatchers
+	s.muDynamicTAGAWSFetchers.Unlock()
+
+	azureSyncMatchers, err := s.accessGraphAzureFetchersFromMatchers(matchers, dc.GetName())
```

Comment:
From @marcoandredinis in https://github.com/gravitational/teleport/pull/48843#discussion_r1888364521
> Can we rename the `s.accessGraphFetchersFromMatchers(` to be explicit about AWS Matchers for AccessGraph?
> 
> So, from `accessGraphAzureFetchersFromMatchers` to `accessGraphAWSFetchersFromMatchers`